src/main/java/io/netty/incubator/codec/http3/Http3RequestStreamValidationHandler.java

```diff
@@ -85,14 +86,39 @@ private State checkState(boolean inbound, State state, Http3RequestStreamFrame f
                         expectedLength = length;
                     }
                 } else if (!server) {
-                    head = HttpMethod.HEAD.asciiName().equals(headersFrame.headers().method());
+                    clientHeadRequest = HttpMethod.HEAD.asciiName().equals(headersFrame.headers().method());
                 }
-                return State.Started;
-            case Started:
+                if (isInformationalResponse(inbound, headersFrame)) {
+                    return State.Initial;
+                } else {
```

Comment:
nit: you can remove the `else` as you return in the if block

Reply:
Replaced with a ternary